### PR TITLE
Revise Deframer design and implementation

### PR DESCRIPTION
**Pull request by bocchino**
* Revise SDD after design review
* Revise implementation after code review
* Pull in update to `STest/STest/Random/Random.cpp` that causes the random seed to be printed when running tests. This helps with debugging, especially if an error occurs in CI, because the seed is part of the test output.
* Fix https://github.com/nasa/fprime/issues/1296

### CODE REVIEW

**Comment by timcanham** [comment]
@bocchino We are looking at an older compiler for a project. Any chance you could replace your `const auto` statements with the actual type?

**Comment by bocchino** [comment]
> Any chance you could replace your const auto statements with the actual type?

Sure. Is the issue just `const auto` or any use of `auto`?

**Comment by timcanham** [comment]
Any use of auto, since the compiler we are trying to use is C++98. We have workarounds for other things, but not for `auto.`

**Comment by bocchino** [comment]
Done.

File: Svc/Deframer/Deframer.cpp [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        U8 *const packetData = packetBuffer.getData();
+        const U32 packetSize = packetBuffer.getSize();
+        switch (packetType) {
+            // Handle a command packet
+            case Fw::ComPacket::FW_PACKET_COMMAND: {
+                // Allocate a com buffer on the stack
+                Fw::ComBuffer com;
+                // Copy the contents of the packet buffer into the com buffer
+                status = com.setBuff(packetData, packetSize);
+                if (status == Fw::FW_SERIALIZE_OK) {
```
Comment (LeStarch) [inline Svc/Deframer/Deframer.cpp:138]: We don't check that it is connected here?  Is the deframer required to have a command connection? We specifically check the connection on the buffer port.

That is: this component can be used w/o buffer connections, but cannot be used w/o comm buffer connections.  Is this asymmetry ok?

**Comment by LeStarch** [comment]
@arizvi786 @kevin-f-ortega  I left some questions here, but I'll defer to you both for the final review.

File: Svc/Deframer/Deframer.cpp [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+}
+
+void Deframer ::schedIn_handler(
+    const NATIVE_INT_TYPE portNum,
+    NATIVE_UINT_TYPE context
+) {
+    // Check for data
+    Fw::Buffer buffer(m_pollBuffer, sizeof(m_pollBuffer));
+    Drv::PollStatus status = framedPoll_out(0, buffer);
+    if (status == Drv::PollStatus::POLL_OK) {
```
[the remark this answers is not in the document]
Comment (bocchino) [reply Svc/Deframer/Deframer.cpp]: Fixed!

File: Svc/Deframer/Deframer.cpp (hunk 1 above)
Comment (bocchino) [reply Svc/Deframer/Deframer.cpp:138]: This is how the inherited code was, and the behavior is captured in the requirements. 

https://github.com/bocchino/fprime/blob/deframer-design-code/Svc/Deframer/docs/sdd.md#3-requirements

See SVC-DEFRAMER-011.

[the remark this answers is not in the document]
Comment (bocchino) [reply Svc/Deframer/Deframer.cpp:138]: See above.

File: Svc/Deframer/docs/img/.fpv-env [hunk 3]
```diff
@@ -0,0 +1 @@
+DATA_FOLDER=top/
```
[the remark this answers is not in the document]
Comment (bocchino) [reply Svc/Deframer/docs/img/.fpv-env:1]: Yes, this file is needed to make the visualization tool work.

[the remark this answers is not in the document]
Comment (bocchino) [reply Svc/Deframer/docs/img/.fpv-env:1]: See above.

File: Svc/Deframer/Deframer.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (arizvi786) [reply Svc/Deframer/Deframer.cpp:138]: see above comment

**Review by arizvi786 — APPROVED** [review]
@bocchino looks good, nice work!

**Pull request merged**